source/AI.cpp

```diff
@@ -1978,7 +1981,17 @@ void AI::PrepareForHyperspace(Ship &ship, Command &command)
 	bool isJump = (ship.GetCheapestJumpType(ship.GetTargetSystem()).first == Ship::JumpType::JumpDrive);
 
 	Point direction = ship.GetTargetSystem()->Position() - ship.GetSystem()->Position();
-	if(!isJump && scramThreshold)
+	double squaredDeparture =  isJump ?
+		ship.GetSystem()->JumpDepartureDistance() * ship.GetSystem()->JumpDepartureDistance() :
+		ship.GetSystem()->HyperDepartureDistance() * ship.GetSystem()->HyperDepartureDistance();
+	squaredDeparture += SAFETY_OFFSET;
+	if(ship.Position().LengthSquared() < squaredDeparture)
+	{
+		// Add a constant to the squared length to be sure to not divide by zero.
+		Point closestDeparturePoint = ship.Position() * (squaredDeparture / (ship.Position().LengthSquared() + ZERO_OFFSET));
```

Comment:
The division by zero is now fixed, but if ship.Position() is (0,0), then the multiplication will still not result in another position than 0,0.

Reply:
you are right...

Reply:
fixed